### PR TITLE
ci: migrate from codefresh to github workflows

### PR DESCRIPTION
**Pull request by tmac1973**
***Issue***: 
https://lacework.atlassian.net/browse/GROW-2760

***Description:***
Provides workflows for testing and release to replace codefresh

### CODE REVIEW

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -2,8 +2,8 @@ name: Nightly Build
 
 on:
   workflow_dispatch:
```
Comment (ipcrm) [inline .github/workflows/nightly.yml:4]: all of these have workflow_dispatch; 

This is required for recovery purposes, correct?

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -72,14 +72,31 @@ jobs:
           make prepare
           make clean-test
           make integration-test
+        env:
+          LW_ACCOUNT: ${{ secrets.LW_ACCOUNT }}.lacework.net
```
Comment (ipcrm) [inline .github/workflows/nightly.yml:76]: secret is shortname? (I assume)

**Review by ipcrm — COMMENTED** [review]
Looks good! couple q's

File: .github/workflows/nightly.yml (hunk 1 above)
Comment (tmac1973) [reply .github/workflows/nightly.yml:4]: This allows us to run the jobs manually if we like, which provides parity with codefresh

File: .github/workflows/test-build.yml [hunk 3]
```diff
@@ -1,9 +1,11 @@
 name: Test Build
 
-on:
+on: 
   workflow_dispatch:
-  # pull_request:
-  # push:
+  pull_request:
```
Comment (ipcrm) [inline .github/workflows/test-build.yml:5]: do we want to run for ANY pull request or only prs targeting main?

File: .github/workflows/test-build.yml [hunk 4]
```diff
@@ -72,14 +74,31 @@ jobs:
...
+          AWS_S3: ${{ secrets.AWS_S3 }}
+          S3_BUCKET_ARN: ${{ secrets.S3_BUCKET_ARN }}
+          LW_PUB_SUB_SUBSCRIPTION: ${{ secrets.LW_PUB_SUB_SUBSCRIPTION }}
 
   slack-notify:
     name: Slack Notify if Failed Tests
     needs: [run-unit-tests, build, run-integration-tests]
     runs-on: ubuntu-latest
-    if: needs.run-unit-tests.result == 'failure' || needs.build.result == 'failure' || needs.run-integration-tests.result == 'failure'
+    if: false # always()
```
Comment (ipcrm) [inline .github/workflows/test-build.yml:98]: I think this should be uncommented 

```suggestion
    if: always()
```

**Review by ipcrm — COMMENTED** [review]
Just a couple questions!

File: .github/workflows/test-build.yml (hunk 3 above)
Comment (tmac1973) [reply .github/workflows/test-build.yml:5]: I don't see a problem running the test for any PR. Do you have a reason to restrict it to main?

Comment (ipcrm) [reply .github/workflows/test-build.yml:5]: Not really tbh; I just know the historic behavior was only build PRs targeting main. But that said, if you don't see an issue with it; neither do I!

Comment (tmac1973) [reply .github/workflows/test-build.yml:5]: changed to main only for PR

**Review by ipcrm — APPROVED** [review]

**Pull request merged**